ONSdigital/design-system
# Add option for additional logo in footer

### What is the context of this PR?

Fixes #3134 
This PR updates the footer component to support an additional logo, allowing for two logos instead of just one. It introduces link param for each logo and offers an alternative layout option to display the logos on opposite sides of the footer. The default layout positions the logos next to each other. Additionally, the implementation is aligned with the header component's multipleLogo feature to ensure consistency across the application. The focus and hover states are also adjusted to avoid extra underlines and to ensure the highlight color wraps only the SVG.

_BREAKING CHANGE_
Services using the poweredBy param with the footer component will now need to update that param to footerLogo object in their config.

### How to review this PR

- Spin up the website and head to the 'footer' component, the example-footer-with-extra-poweredby-logo on the page is the one to look at.
- Check for multiple logos: Verify that the footer displays two logos as expected, with appropriate links set for each logo.
- Review layout options: Confirm that the default layout positions the logos next to each other and test the alternative layout option where the logos are displayed on opposite sides of the footer.
- Test focus and hover states: Ensure that the focus and hover states do not create an extra underline and that the highlight color wraps only the SVG, not its margin.
- run yarn test --testNamePattern="macro: footer" & yarn test-visual

### Checklist

This needs to be completed by the person raising the PR.
